### DISCUSSION TITLE
Will accessibility tests also run automatically when publishing to chromatic?

### DISCUSSION THREAD

**Discussion by Axedyson** [storybookjs/storybook#17966] [Help] [answered]
Right now interaction tests run automatically when publishing components to chromatic in CI, which is awesome as stated [here](https://storybook.js.org/blog/test-component-interactions-with-storybook/):

> Chromatic supports Storybook interaction tests out-of-the-box. It waits for the interaction test to run before capturing the snapshot. This way, you can verify both the visual appearance and the underlying logic for a component in one go. Any test failures get reported via the Chromatic UI.

But what would be even better is if the accessibility tests from `storybook-addon-a11y` would also automatically run when publishing to chromatic avoiding the need for an extra `.test` file. Are there any future plans for this?

**Comment by mririgoyen** [comment]
It's been over a year since this was asked. Any updates from Chromatic on the subject?

**Comment by ghost** [comment]
Reviving this discussion so this topic won't die 👀

**Comment by noranda** [comment]
I'm also looking for something like this. I love the accessibility addon, but there's no automated way to verify all tests pass that I can see. It would be great if this would work with Chromatic! While I understand this is tough to do for all combinations of variants, it sure would be nice! My current workaround is to use the `storybook-addon-variants` addon and just manually check I have no violations for all variants in one story.

**Comment by Zyzle** [comment]
Just chiming in to keep the discussion alive. This is something my team has been looking for, it would be good if chromatic would report on the a11y status of stories along side the interaction and visual tests. Not necessarily to fail builds on but it would be good to see this plugin status for our components without having to manually go through the individual stories

**Reply by noranda** [reply]
> Not necessarily to fail builds on but it would be good to see this plugin status for our components without having to manually go through the individual stories

I love this idea - report on the failed a11y tests, but don't fail the builds. The more stories we build, the more impossible it becomes to manually check every component story for accessibility issues.

**Comment by cahrens** [comment]
I am also interested in this feature.

**Comment by benposnick** [comment]
Also very interested in this feature!

**Comment by drewbrend** [comment]
I would also like this to be built into Chromatic.  I've put together a way to automate it based on this example from storybook, however it leaves a lot to be desired for setup, usability, and debugging.

https://storybook.js.org/blog/automate-accessibility-tests-with-storybook/

Cypress has started offering this, Chromatic really should offer something similar to their offering, it's nice.  https://www.cypress.io/blog/introducing-cypress-accessibility

**Comment by yvan33** [comment]
Any updates on this feature request? It seems natural for Chromatic to provide that feature, given Storybook already provides a way to run [a11y tests using the test-runner](https://storybook.js.org/docs/writing-tests/accessibility-testing#automate-accessibility-tests-with-test-runner).

**Comment by shilman** [comment, marked as answer]
YES!! Key updates here:
- [Accessibility Addon Refresh](https://storybook.js.org/blog/preview-the-new-accessibility-addon/)
- [Chromatic Accessibility Regression Testing](https://www.chromatic.com/blog/sneak-peek-accessibility-regression-testing/)
